hr_employee_document/models/ir_rule.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Tecnativa - Víctor Martínez
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, models, tools
+from odoo.osv import expression
+from odoo.tools import config
+
+
+class IrRule(models.Model):
+    _inherit = "ir.rule"
+
+    def _get_hr_employee_document_group(self):
+        return "hr.group_hr_user"
+
+    @api.model
+    @tools.conditional(
+        "xml" not in config["dev_mode"],
+        tools.ormcache(
+            "self.env.uid",
+            "self.env.su",
+            "model_name",
+            "mode",
+            "tuple(self._compute_domain_context_values())",
+        ),
+    )
+    def _compute_domain(self, model_name, mode="read"):
+        """We need to add for security purposes an extra domain in the hr.employee
+        model to restrict only the user's employees when search employee attachments."""
+        res = super()._compute_domain(model_name, mode=mode)
+        user = self.env.user
+        if model_name == "hr.employee" and not self.env.su:
```

Comment:
Shouldn't this also exclude HR managers?

Reply:
Now the extra domain is not added to the rule if the user has the Employees > Officer permission (this is the same behavior as the module currently has).

Reply:
But our use case wants to exclude Officers. We want this only for HR managers, not officers.